tripy/docs/conf.py

```diff
@@ -178,40 +187,40 @@ def process_docstring(app, what, name, obj, options, lines):
                     ":returns:" in doc
                 ), f"For: {obj}, return value is not documented. Please ensure you've included a `Returns:` section"
 
-    # new docstring logic:
-    # first figure out if we should it is the new docstring
+    # New docstring logic:
+    # First figure out if we should it is the new docstring.
     if name.split(".")[-1] in TYPE_VERIFICATION.keys():
         cleaned_name = name.split(".")[-1]
         add_text_index = -1
         for index, block in enumerate(blocks):
             if re.search(r".. code-block::", block):
-                type_dict = TYPE_VERIFICATION[cleaned_name][1]["types"]
+                type_dict = TYPE_VERIFICATION[cleaned_name].dtypes
                 blocks.insert(index, "Type Constraints:")
                 index += 1
+                # Add the dtype constraint name and the dtypes that correlate. Remove excluded dtypes.
                 for type_name, dt in type_dict.items():
-                    blocks.insert(index, f"    - {type_name}: " + ", ".join(dt))
+                    blocks.insert(
+                        index,
+                        f"    - **{type_name}**: :class:`" + "`, :class:`".join(set(dt) - {"int4", "float8"}) + "`",
+                    )
                     index += 1
                 blocks.insert(index, "\n")
                 break
             if re.search(r":param \w+: ", block):
-                add_text_index = re.search(r":param \w+: ", block).span()[1]
                 param_name = re.match(r":param (\w+): ", block).group(1)
-                blocks[index] = (
-                    block[0:add_text_index]
-                    + "[dtype="
-                    + TYPE_VERIFICATION[cleaned_name][2][param_name]
-                    + "] "
-                    + block[add_text_index:]
-                )
+                # Add dtype constraint to start of each parameter description.
+                if TYPE_VERIFICATION[cleaned_name].dtype_constraints.get(param_name, None):
+                    add_text_index = re.search(r":param \w+: ", block).span()[1]
+                    blocks[index] = (
+                        f"{block[0:add_text_index]}[dtype=\ **{TYPE_VERIFICATION[cleaned_name].dtype_constraints[param_name]}**\ ] {block[add_text_index:]}"
+                    )
             if re.search(r":returns:", block):
-                add_text_index = re.search(r":returns:", block).span()[1] + 1
-                blocks[index] = (
-                    block[0:add_text_index]
-                    + "[dtype="
-                    + list(TYPE_VERIFICATION[cleaned_name][1]["returns"].values())[0]["dtype"]
-                    + "] "
-                    + block[add_text_index:]
-                )
+                if TYPE_VERIFICATION[cleaned_name].dtype_constraints.get(dtype_info.RETURN_VALUE, None):
```

Comment:
We already check that return values are documented, so you can just keep the inner condition here.

Reply:
This check checks if the parameter has a dtype constraint, if not it leaves as is.

Reply:
I mean you can remove the `if re.search(r":returns:", block):`

Reply:
But that needs to be there know when we are on the return line so that it can place the dtype info for returns on the line.

Reply:
Oh right, ok. I thought the block was the whole text block, but it's actually just the current line.